tests/unit/parallel/executors/shared_parallel_executor.cpp

```diff
@@ -32,6 +33,17 @@ struct shared_parallel_executor
     {
         return hpx::async(std::forward<F>(f));
     }
+
+    template <typename F>
+    typename std::remove_reference<
+        typename hpx::util::result_of<
+            typename hpx::util::decay<F>::type()
+        >::type
+    >::type
```

Comment:
I tried solving it in the trait, but was not able to do it there. If you have an idea how this could be formulated, I'd appreciate any ideas.